src/placement/ring_map.c

```diff
@@ -1253,8 +1253,10 @@ ring_obj_find_reint(struct pl_map *map, struct daos_obj_md *md,
 			return rc;
 		rc = pl_obj_layout_alloc(rop.rop_grp_size, rop.rop_grp_nr,
 				&reint_layout);
-		if (rc)
+		if (rc) {
+			D_FREE(layout);
```

Comment:
This seems like something that ought to be handled internally to pl_obj_layout_alloc, no?  Using D_FREE is odd here otherwise because the allocation happens inside of pl_obj_layout_alloc which has a corresponding pl_obj_layout_free.  But if the function fails, it should not return allocated memory.